src/libos/src/process/do_vfork.rs

```diff
@@ -101,6 +99,48 @@ pub fn is_vforked_child_process() -> bool {
     })
 }
 
+fn vfork_save_file_table(current: &ThreadRef) -> Result<()> {
+    let parent_pid = current.process().pid();
+    let mut vfork_file_tables = VFORK_PARENT_FILE_TABLES.lock().unwrap();
+    let parent_file_table = {
+        let mut current_file_table = current.files().lock();
+        let new_file_table = current_file_table.clone();
+        // FileTable contains non-cloned struct, so here we do a memory replacement to use new
+        // file table in child and store the original file table in TLS.
+        mem::replace(&mut *current_file_table, new_file_table)
+    };
+
+    // Insert the parent file table. The key shouldn't exist because there must be only one thread doing the vfork and save the file table for current process.
+    let ret = vfork_file_tables.insert(parent_pid, parent_file_table);
+    debug_assert!(ret.is_none());
+    Ok(())
+}
+
+fn vfork_stop_all_child_thread(current: &ThreadRef) {
+    // stop all other child threads
+    loop {
+        let child_threads = current.process().threads();
+        let running_threads = child_threads
```

Comment:
`let running_threads_count = child_threads.iter().filter(...).map(|thread| ....).count()`

Reply:
Done. Thanks for your review.